test/unit/main_test.ts

```diff
@@ -40,3 +62,137 @@ describe('Recognizes invalid TypeScript inputs', () => {
+
+  it('supports visibility modifiers', () => {
+    expectTranslate('class X { private _x; }').to.equal('class X { private _x; }');
+    expectTranslate('class X { public x; }').to.equal('class X { public x; }');
+    expectTranslate('class X { protected x; }').to.equal('class X { protected x; }');
+  });
+
+  it('supports static fields', () => {
+    expectTranslate('class X { static x: number = 42; }')
+        .to.equal('class X { static x: number = 42; }');
+  });
+
+  it('should suppport creating an instance of a class',
+     () => { expectTranslate('x = new X("blah");').to.equal('x = new X("blah");'); });
+});
+
+describe('modules', () => {
+  it('supports modules',
+     () => { expectTranslate('module Validation {}').to.equal('module Validation {}'); });
+  it('supports import equals statements', () => {
+    expectTranslate('import someMod = require("someModule");')
+        .to.equal('import someMod = require("someModule");');
+  });
+  it('supports import from statements', () => {
+    expectTranslate('import * as ts from "typescript";')
+        .to.equal('import * as ts from "typescript";');
+    expectTranslate('import {x} from "./y";').to.equal('import {x} from "./y";');
+    expectTranslate('import {Foo as Bar} from "baz";').to.equal('import {Foo as Bar} from "baz";');
+    expectTranslate('import {} from "baz";').to.equal('import {} from "baz";');
+  });
+});
+
+describe('exports', () => {
+  it('supports export equals statements', () => {
+    expectTranslate('export = LettersOnlyValidator;').to.equal('export = LettersOnlyValidator;');
+  });
+  it('allows class exports',
+     () => { expectTranslate('export class Polygons {}').to.equal('export class Polygons {}'); });
+  it('allows variable exports',
+     () => { expectTranslate('export var x = 10;').to.equal('export var x = 10;'); });
+  it('allows export declarations',
+     () => { expectTranslate('export * from "X";').to.equal('export * from "X";'); });
+  it('allows named export declarations',
+     () => { expectTranslate('export {a, b} from "X";').to.equal('export {a, b} from "X";'); });
+  it('allows renamed exports', () => {
+    expectTranslate('export {Foo as Bar} from "baz";').to.equal('export {Foo as Bar} from "baz";');
+  });
+});
+
+describe('expressions', () => {
+  it('does math', () => {
+    expectTranslate('1 + 2').to.equal('1 + 2');
+    expectTranslate('1 - 2').to.equal('1 - 2');
+    expectTranslate('1 / 2').to.equal('1 / 2');
+    expectTranslate('1 * 2').to.equal('1 * 2');
+    expectTranslate('1 % 2').to.equal('1 % 2');
+    expectTranslate('x++').to.equal('x++');
+    expectTranslate('x--').to.equal('x--');
+    expectTranslate('++x').to.equal('++x');
+    expectTranslate('--x').to.equal('--x');
+  });
+  it('shows correctly appends "$mangled" to an identifier of a property access expression',
+     () => { expectTranslate('Math.random();').to.equal('Math.random$mangled();'); });
```

Comment:
You wanted to add more tests for the recursion vs non-recursion case here, didn't you? e.g. some mangled thing in a for loop, or in a class definition?